time_sequence_prediction/train.py

```diff
@@ -39,28 +39,41 @@ def forward(self, input, future = 0):
 if __name__ == '__main__':
     parser = argparse.ArgumentParser()
     parser.add_argument('--steps', type=int, default=15, help='steps to run')
+    parser.add_argument('--no-cuda', action='store_true', default=False,
+                        help='disables CUDA training')
+    parser.add_argument('--no-mps', action='store_true', default=False,
+                        help='disables macOS GPU training')
     opt = parser.parse_args()
+    # 
+    # Use GPU to train?
+    use_cuda = not opt.no_cuda and torch.cuda.is_available()
+    use_mps = not opt.no_mps and torch.backends.mps.is_available()
+    if use_cuda:
+        device = torch.device("cuda")
+    elif use_mps:
+        device = torch.device("mps")
+    else:
+        device = torch.device("cpu")
     # set random seed to 0
     np.random.seed(0)
     torch.manual_seed(0)
     # load data and make training set
     data = torch.load('traindata.pt')
-    input = torch.from_numpy(data[3:, :-1])
-    target = torch.from_numpy(data[3:, 1:])
-    test_input = torch.from_numpy(data[:3, :-1])
-    test_target = torch.from_numpy(data[:3, 1:])
+    input = torch.from_numpy(data[3:, :-1]).to(device)
```

Comment:
Same problem, you can have one torch tensor on your device and then you can take slices from it in torch as well so avoid these repeated to device calls